### ISSUE TITLE
One reading of what a `url()` is in a text, for every place that has to find one

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#427]
### What minimal example or steps are needed to reproduce the bug?

Half a dozen places in the plugin have to find an address in a text rather than ask about a node a parser already made, and none of them reads what a `url()` is the way CSS does. [#344](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/344) settled how the *name* of an address is read; this is the other half — what a `url()` is once you have to look for it.

Five faults, all measured, all present on `main` today.

**A `/*` written inside an address opens a comment to the scan that finds comments.** The scan runs it to a `*/` standing anywhere later in the file, or to the end of the file where there is none, so everything behind the asterisk is taken for the text of a comment and every rule reading that scan passes it over:

```js
findCommentSpans(`url(http://x/*.png) 1px`)   // [{ start: 9, end: 23, isInline: true }] — a comment that is none
```

CSS reads no comment inside a `url()` token at all. `lightningcss` answers `BadUrl("a /* ")` to `a { background: url(a /* ) */ b.png) }` and `Unexpected token CloseParenthesis` to `a { background: url(/*)*/a) }` — the token closed at the parenthesis standing inside what looks like a comment, both times.

**`max-line-length` finds an address with a pattern, and the pattern is wrong four ways.** `EVERY_URL_CONTENT` is `/url\(\s*(\S.*\S)\s*\)/gui`: it searches a line for `url(` with no boundary in front of it and runs greedily to the last parenthesis on the line. Smallest `max` at which each line goes quiet:

| line | reported at | should be |
|---|---|---|
| `a { background: url(a.png) no-repeat, linear-gradient(red, blue); }` | 24 | 62 |
| `@import url(a.css) screen and (orientation:landscape);` | 14 | 49 |
| `a { background: image-url(aaaaaaaaaaaa); }` | never | 38 |
| `.m(url(a,b)) { c: 2px; }` | 23 | 24 |
| `a { background: url("http://x/y.png") }` inside `/* … */` | 25 | 37 |

So a line carrying an address and anything parenthesised behind it is undercounted; a call whose name merely ends in those three letters has its arguments taken off the count as though they were an address; a `url(` written inside a comment or inside a quoted string is read as a call at all. One goes the other way: `(\S.*\S)` needs two characters, so an address of one is never taken off — `bbbbbbbbbbbbbbbb: url(a);` is reported at 26 where `url(ab)` is quiet at 25.

**`max-line-length` also reads an address by four characters**, which is the sixth place of #344 and the one that could not be fixed there: `u\rl(http://…)` is counted where `url(http://…)` is not.

**Nothing tells a url token from the `url()` function.** Whitespace and then a quotation mark after the name open the function, whose arguments are code; anything else opens a token, whose text is the address entire. Sass reads `url("x" /* c */, f(1))` as `url("x", f(1))` — a real comment among real arguments — and leaves the comma of a nested `URL(a,b)` while spacing the comma of a nested `f(a,b)`. Neither Sass nor Less accepts a double slash among those arguments, both answering `expected ")"`.

**`URL_CALL_AT_END` asks "does a name stand in front of this address" with a complement of ASCII**, which is [#398](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/398) and belongs to the same repair.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/max-line-length": 20
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json
```

### What did you expect to happen?

One reading of what a `url()` is in a text, asked wherever an address has to be found: the name read as CSS spells it, the token told from the function, a comment inside a token read as characters of the address, and a `url(` inside a comment read as no address at all.

### What actually happened?

Each place looks for an address its own way, and every way is wrong somewhere.

### What is the cause?

Comments and addresses cannot be found by two scans, because each is the other's exception: the double slashes of a protocol open no comment, and a `url(` written inside a comment opens no address. `findCommentSpans` carries the only walk that knows both — it has `skipUrl` — and it is not asked for the addresses it steps over, so every other place looks for them with a pattern or with a scan that knows nothing of them.

### Do you have a proposal to fix the bug?

One walk, two answers. Have the walk behind `findCommentSpans` collect the addresses it steps over and hand them back beside the comments, tell a token from the function while it is there, and put `max-line-length` and `endsWithInlineComment` on that. A branch that did all of this exists and is measured — [`bf-344-wide`](https://github.com/stylelint-stylistic/stylelint-stylistic/tree/bf-344-wide), pushed for reference and kept off `main`: by the time it worked it was answering five questions at once, and #344 turned out to be a smaller one that needed none of them. Read it for the shape of the answer and for the measurements, not as a patch — six rounds of review turned up defect after defect in it, and its last version is the sixth attempt rather than a settled one.
